lib/sdcadm.js

```diff
@@ -1413,42 +1487,48 @@ SdcAdm.prototype.getCandidateImages = function getCandidateImages(opts, cb) {
              * updates. So we'll look for images published after 'A'
              * (including 'A' to allow same-image updates for dev/testing).
              */
-            common.sortArrayOfObjects(currImgs, ['published_at']);
-            var name = self.config.imgNameFromSvcName[opts.service.name];
-            if (!name) {
-                var msg = format('do not know image name for service "%s"',
-                        opts.service.name);
-                self.log.info(msg);
-                opts.progress(msg);
-                next();
-                return;
-            }
-            var filter = {
-                name: name,
-                marker: (currImgs.length > 0 ?
-                    currImgs[0].published_at : undefined)
-            };
 
-            self.log.debug({filter: filter},
+            common.sortArrayOfObjects(currImgs, ['published_at']);
+            let marker = (currImgs.length > 0 ?
+                currImgs[0].published_at : undefined);
+            self.log.debug({marker: marker, validImgNames: validImgNames},
                 'getCandidateImages: getCandidates');
-            self.updates.listImages(filter, function (uErr, followingImgs) {
-                if (uErr) {
-                    next(uErr);
+
+            var allCandidates = [];
+            vasync.forEachPipeline({
+                inputs: validImgNames,
+                func: function listOneName(name, nextName) {
+                    self.updates.listImages({
+                        name: name,
+                        marker: marker
+                    }, function (uErr, candidates) {
+                        if (uErr) {
+                            nextName(uErr);
+                            return;
+                        }
+
+                        // Sanity check that IMGAPI properly returns images with
+                        // the given name (e.g. TOOLS-745).
+                        candidates = candidates.filter(function (i) {
```

Comment:
Maybe thinking ahead here, but won't this mean we can never upgrade a mantav1-binder to a mantav2-binder? Is the upgrade path going to be done by something other than sdcadm?
(then see my next question)

Reply:
First, no Triton should *ever* have a binder using the "mantav1-nameservice" image name.
Also, no this won't break (at least I don't believe so), if the existing image name for a given instance isn't one of the current valid/blessed image names.